.github/workflows/reusable-build-upload.yaml

```diff
@@ -0,0 +1,190 @@
+name: "Publish specific rust package"
+
+on:
+  workflow_call:
+    inputs:
+      # Package related variables
+      package-name:
+        description: "The package name to use (ex: gz-srv)"
+        type: string
+        required: true
+        default: ""
+
+      package-version:
+        description: "The package version to use (ex: 0.1.2)"
+        type: string
+        required: true
+        default: ""
+
+      package-git-tag:
+        description: "the release tag name (ex: gz-srv-v0.1.2)"
+        type: string
+        required: true
+        default: ""
+
+      # CF Bucket related variables
+      cf-bucket-name:
+        description: "The CF bucket name to use"
+        required: true
+        type: string
+
+      cf-config-bucket-root-key:
+        description: "The root key to be used for accessing the configs. (ex: `test-root-key` puts releases in `test-root-key/*`)"
+        required: true
+        type: string
+
+    secrets:
+      github-token:
+        description: "The github token to use to do the tag updates"
+        required: true        
+      cf-endpoint-url:
+        description: "The endpoint URL of the CF bucket"
+        required: true
+      cf-bucket-access-key-id:
+        description: "The CF bucket access key id"
+        required: true
+      cf-bucket-secret-access-key:
+        description: "The CF bucket secret access key"
+        required: true
+
+jobs:
+  build-and-upload:
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.github-token }}
+          
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.76.0
+          targets: wasm32-unknown-unknown
+
+      - name: Build artifacts for ${{ inputs.package-name }}-v${{ inputs.package-version }}
+        run: |
+          echo "Building wasm for '${{ inputs.package-name }}-v${{ inputs.package-version }}'";
+          # 21.1.1 is the latest rust 1.76.0 compatible version, the version after is 21.2.0 which is compatible with 1.79.0 (different than the workspace's version)
+          cargo install --locked soroban-cli --version 21.1.1 --features opt
```

Comment:
the Stellar Testnet upgraded to v22 recently, the version should be updated accordingly. ref: https://github.com/axelarnetwork/axelar-cgp-soroban/pull/64

Reply:
I think the release candidate will be released on Nov 17, we can open another pr then and shift off the "rc" version.